node/network/collator-protocol/src/collator_side.rs

```diff
@@ -138,16 +229,13 @@ struct State {
 /// or the relay-parent isn't in the active-leaves set, we ignore the message
 /// as it must be invalid in that case - although this indicates a logic error
 /// elsewhere in the node.
-async fn distribute_collation<Context>(
-	ctx: &mut Context,
+async fn distribute_collation(
+	ctx: &mut impl SubsystemContext<Message = CollatorProtocolMessage>,
```

Comment:
Also, we please not carpet-format things and mix it with other changes :pray:? AFAIK different subsystems atm use both variants of this syntax.

Reply:
Nope. When I fix stuff, I'm doing it everywhere. A better solution would anyway to implement all these functions for the state struct. It makes no sense that all these functions are free standing, this is not C.

Reply:
> It makes no sense that all these functions are free standing, this is not C.
 
This is a stylistic difference I strongly disagree with you on. Rust excels when used in a functional style, especially in async/await.

Reply:
Other sub systems are exactly written in the style I mentioned. I also see no benefit to always pass the same two parameters in `state` and `context` to every function.

And async/await doesn't require that you only use free standing functions.